webclient/src/websocket/commands/SessionCommands.ts

```diff
@@ -110,6 +115,31 @@ export class SessionCommands {
     });
   }
 
+  static requestPasswordSalt(): void {
+    const options = webClient.options as unknown as RequestPasswordSaltParams;
+
+    const registerConfig = {
+      ...webClient.clientConfig,
+      userName: options.user,
+    };
+
+    const CmdRequestPasswordSalt = webClient.protobuf.controller.Command_RequestPasswordSalt.create(registerConfig);
+
+    const sc = webClient.protobuf.controller.SessionCommand.create({
+      ".Command_RequestPasswordSalt.ext" : CmdRequestPasswordSalt
+    });
+
+    webClient.protobuf.sendSessionCommand(sc, raw => {
+      if (raw.responseCode === webClient.protobuf.controller.Response.ResponseCode.RespOk) {
+        const passwordSalt = raw[".Response_PasswordSalt.ext"].passwordSalt;
+        SessionCommands.login(passwordSalt);
+      } else if (raw.responseCode === webClient.protobuf.controller.Response.ResponseCode.RespRegistrationRequired) {
+          SessionCommands.updateStatus(StatusEnum.DISCONNECTED, "Login failed: Registration Required");
```

Comment:
I'm a bit confused, does RespRegistrationRequired mean the user is not registered at all, or is awaiting account verification?

Reply:
respregistration required means that this user is not recognized or eligible for login (they do not have a password in the database) and is not allowed to log in unregistered.

Reply:
this check here does not check if the account is awaiting verification, I didn't test that yet

Reply:
So really the message we want here is "Invalid Username/Password". For security, you dont want to differentiate between username and password issues for failed logins, unless it's a special case like AccountActivationRequired or something, in which case the issue isnt with the username/password at all, but with the account status itself.

Reply:
In other words, if someone attempts to login with an account that doesnt exist, we shouldnt expose that information to them. To the user, attempting to login with an account that doesnt exist is identical to attempting to login with a wrong password.

My initial thought is maybe the server should just return a dummy salt for unknown users.

Reply:
I fully agree with you, yet this is the way it works currently on login, you can ask the server which account names exist anyway if you try to register their names, I don't think changing this is a priority